client/packages/host/src/components/Sync/SyncModal.tsx

```diff
@@ -137,203 +147,160 @@ export const SyncModal = ({
     syncStatus?.summary?.durationInSeconds || 0
   );
 
+  const getSyncStatusMessage = (): string => {
+    if (syncStatus?.isSyncing === true) {
+      return t('sync-info.syncing');
+    } else {
+      return numberOfRecordsInPushQueue
+        ? t('label.records-to-push', { count: numberOfRecordsInPushQueue })
+        : t('label.no-records-to-push');
+    }
+  };
+
+  const lastSuccessfulSyncMessage = (
+    date: Date,
+    durationAsDate: Date
+  ): string => {
+    // If the date is today, display the time, otherwise display the date
+    let today = new Date();
+    let lastSuccessfulSyncTime;
+    if (date?.toDateString() === today.toDateString()) {
+      lastSuccessfulSyncTime = localisedTime(date);
+    } else {
+      lastSuccessfulSyncTime = localisedDate(date);
+    }
```

Comment:
```suggestion
     const today = new Date();
    const lastSuccessfulSyncTime =
      date?.toDateString() === today.toDateString()
        ? localisedTime(date)
        : localisedDate(date);
```

Always a mission flipping between rust and TS, but we do our best to use `const` in TS, and use patterns that allow us to initialise the const with the correct value, rather than setting an undefined variable with an if/else block.

Especially in React, working with mutable values can lead to some nasty and hard to trace bugs, so we try be in the habit of immutable values wherever possible. Using const also means I can use `lastSuccessfulSyncTime` with confidence - I don't need to read through the rest of the function to make sure there aren't other conditions where it could get reassigned.